src/algebra/order/hom/equiv.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2022 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández, Jiale Miao
+-/
+import algebra.order.hom.basic
+import analysis.special_functions.pow
+
+/-!
+# `equiv` on nonnegative morphisms
+
+This file defines the equivalence of two nonnegative morphisms.
+
+Two nonnegative morphisms `f, g` on `R` are equivalent if there exists a positive constant
+`c` such that for all `x ∈ R`, `(f x) ^ c = g x`.
+-/
+
+namespace nonneg_hom
+
+/-- Two nonnegative morphisms `f, g` on `R` are equivalent if there exists a positive constant
+  `c` such that for all `x ∈ R`, `(f x) ^ c = g x`. -/
+def equiv {R F : Type*} [nonneg_hom_class F R ℝ] (f g : F) :=
+∃ c : ℝ, 0 < c ∧ (λ x : R, (f x) ^ c) = g
```

Comment:
I don't know whether `∃ c : ℝ, 0 < c ∧ ∀ x, (f x) ^ c = g x` is a better definition (I think that this is what would be written in the textbooks, but I'm not sure if that matters).